game.sy

```diff
@@ -381,18 +381,51 @@ start :: fn {
 
     l_set_window_size' (800, 800) //TODO(gu): set on startup so the window doesn't change size during start
 
+    volume := 0
+    max_volume := 10
+    vol_cooldown := 0.0
+
     bind_keys'
     loop !exit {
         l_update'
         l_render'
 
+        input := poll_input'
         if show_main_menu {
             main_menu.process_input' menu.from_input' poll_input'
         } else {
-            set_input' poll_input'
+            set_input' input
             process_player_input' l_delta'
         }
 
+        if vol_cooldown == 0.0 {
+            if input.dir[0] > 0.5 {
+                volume += 1
+                vol_cooldown = 0.2
+            } else if input.dir[0] < -0.5 {
+                volume -= 1
+                vol_cooldown = 0.2
+            }
+            volume = clamp' volume, 0, max_volume
+        }
```

Comment:
Isn't this the same as your ticker logic?

Reply:
It is! Nice catch